src/Sentry.Unity/ScreenshotAttachment.cs

```diff
@@ -0,0 +1,107 @@
+using System;
+using System.IO;
+using Sentry;
+using Sentry.Extensibility;
+using UnityEngine;
+using UnityEngine.Rendering;
+
+
+namespace Sentry.Unity
+{
+    internal class ScreenshotAttachment : Attachment
+    {
+        public ScreenshotAttachment(IAttachmentContent content)
+            : base(AttachmentType.Default, content, "screenshot.jpg", "image/jpeg") { }
+    }
+
+    internal class ScreenshotAttachmentContent : IAttachmentContent
+    {
+        private readonly SentryMonoBehaviour _behaviour;
+        private readonly SentryOptions _options;
+
+        public ScreenshotAttachmentContent(SentryOptions options, SentryMonoBehaviour behaviour)
+        {
+            _behaviour = behaviour;
+            _options = options;
+        }
+
+        public Stream GetStream()
+        {
+            if (!_behaviour.MainThreadData.IsMainThread())
+            {
+                _options.DiagnosticLogger?.LogDebug("Won't capture screenshot because we're not on the main thread");
+                return new MemoryStream();
```

Comment:
Today this will render a broken UI preview because of a 0 byte attachment:

![image](https://user-images.githubusercontent.com/1633368/160937405-ad1cf751-49f8-4aef-91c9-d78e9754b882.png)


No tracking issue I don't think but we should figure something out

Reply:
With the current options we have in dotnet I don't think we can avoid sending an empty attachment.

I guess the UI needs to be fixed - The [specification](https://develop.sentry.dev/sdk/features/#screenshots) says: "In this case, a 0 byte attachment will be included. In that case, Sentry will not show a screenshot preview."

Reply:
https://github.com/getsentry/sentry-dotnet/pull/1557
With this PR up you can instead of logging with debug log, throw an exception with the above message that the attachment will not be linked to the current event.

Reply:
Maybe if we could change `public Stream GetStream()` to `public Stream? GetStream()` then we wouldn't even need to log an exception in the above case, because we can recognize it's not going to be available. But because `IAttachmentContent` is a public interface, any change would be breaking...

Reply:
What about returning a custom `NoStream()` just for this case and recognizing it while adding the attachment?

Reply:
After a finding today that CaptureScreenshotAsTexture() only throws in development build and crashes the app in a release one, I've brought back the manual check for the main thread and thrown an exception explicitly. Starting the text with "Sentry:" seems to cause some part of the SDK (I think I've seen that somewhere but can't remember) to not report that exception.